portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java

```diff
@@ -0,0 +1,82 @@
+package com.google.sps.servlets;
+
+import static org.apache.lucene.util.SloppyMath.haversinMeters;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import com.google.sps.data.Grid;
+import com.google.sps.data.Location;
+import com.google.sps.data.Marker;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jsoup.Jsoup;
+import org.jsoup.safety.Whitelist;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.lang.*;
+
+/** Handles fetching and saving markers data. */
+@WebServlet("/numberOfReports")
+public class NumberOfReportsServlet extends HttpServlet {
+  private static final Gson gson = new Gson();
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  //List<Grid> reportsInGrids = new ArrayList<Grid>();
+  double lat;
+  double lng;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    MarkerServlet markerServlet = new MarkerServlet();
+    List<Marker> markers = markerServlet.getMarkers(request);
+    List<Grid> reportsInGrids = new ArrayList<Grid>();
+    reportsInGrids = getNumOfReportsforAll(reportsInGrids, markers);
+    String json = gson.toJson(reportsInGrids); //returns the number of reports for all grids
+    response.setContentType("application/json");
+    response.getWriter().println(json);    
+  }
+
+  //for each marker in the arraylist, get num of reports
+  public List<Grid> getNumOfReportsforAll(List<Grid> reportsInGrids, List<Marker> markers){
+    for(Marker marker : markers){
```

Comment:
I'm not sure what this is trying to do. it should just be 
`return markers.stream().filter(marker => marker.getGridRow() == gridrow && maker.getGridCol() == gridcol).count();`
  
you will need to delete/backfill any markers that don't have grids on them. Or you can change `getMarkers()` to not return any markers that doesn't have gridrow and gridcol